tests/testthat/test-5getExperimentCrossAssociation.R

```diff
@@ -259,27 +260,66 @@ test_that("getExperimentCrossCorrelation", {
     # Values should be the same
     expect_equal(round(result, 4), round(ref, 4))
     
-    mae <- mae[1:10, 1:10]
+    mae_sub <- mae[1:10, 1:10]
     # Test that output is in correct type
     expect_true( is.data.frame(suppressWarnings(
-        testExperimentCrossCorrelation(mae, p_adj_threshold = NULL))) )
+        testExperimentCrossAssociation(mae_sub, p_adj_threshold = NULL))) )
     expect_true( is.data.frame(suppressWarnings(
-        getExperimentCrossCorrelation(mae, test_significance = TRUE, p_adj_threshold = NULL))) )
-    expect_true( is.data.frame(getExperimentCrossCorrelation(mae)) )
+        getExperimentCrossAssociation(mae_sub, test_significance = TRUE, p_adj_threshold = NULL))) )
+    expect_true( is.data.frame(getExperimentCrossAssociation(mae_sub, show_warnings = F)) )
     # There should not be any p-values that are under 0
     expect_true( is.null(suppressWarnings(
-        testExperimentCrossCorrelation(mae, p_adj_threshold = 0))) )
+        testExperimentCrossAssociation(mae_sub, p_adj_threshold = 0))) )
     # Test that output is in correct type
     expect_true( is.list(suppressWarnings(
-        testExperimentCrossCorrelation(mae, mode = "matrix", 
+        testExperimentCrossAssociation(mae_sub, mode = "matrix", 
                                           p_adj_threshold = NULL))) )
     expect_true( is.list(suppressWarnings(
-        getExperimentCrossCorrelation(mae, test_significance = TRUE, 
+        getExperimentCrossAssociation(mae_sub, test_significance = TRUE, 
                                       mode = "matrix", p_adj_threshold = NULL))) )
-    expect_true( is.matrix(getExperimentCrossCorrelation(mae, mode = "matrix")) )
+    expect_true( is.matrix(getExperimentCrossAssociation(mae_sub, mode = "matrix", 
+                                                         show_warnings = F)) )
+    
     # There should not be any p-values that are under 0
-    expect_true( is.null(suppressWarnings(
-        testExperimentCrossCorrelation(mae, p_adj_threshold = 0, mode = "matrix"))) )
+    expect_true( is.null(
+        testExperimentCrossAssociation(mae_sub, p_adj_threshold = 0, mode = "matrix", show_warnings = FALSE)) )
     
+    # When correlation between same assay is calculated, calculation is made faster
+    # by not calculating duplicates
+    cor <-  testExperimentCrossAssociation(mae, experiment1 = 1, experiment2 = 1, 
+                                           show_warnings = F)
+    # Get random variables and test that their duplicates are equal
+    for(i in 1:10 ){
+        random_var1 <- sample(cor$Var1, 1)
+        random_var2 <- sample(cor$Var1, 1)
+        expect_equal(as.numeric(cor[cor$Var1 == random_var1 & cor$Var2 == random_var2, c("cor", "pval", "p_adj")]),
+                     as.numeric(cor[cor$Var1 == random_var2 & cor$Var2 == random_var1, c("cor", "pval", "p_adj")]))
+    }
+    # Test that paired samples work correctly
```

Comment:
Some relevant dissimilarity functions (for instance, Kullback-Leibler divergence) can be asymmetric. 

It is good to check this here somehow but I would suggest to add an argument that defines if we can expect symmetric measure. In general, the users should be able to provide asymmetric measures as well, if they choose so. Unless this causes serious bottlenecks for the implementation.